functions/api/image/tokens/[[index]].tsx

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable import/no-unused-modules */
+import { ImageResponse } from '@vercel/og'
+import React from 'react'
+
+import getColor from '../../../utils/getColor'
+import getNetworkLogoUrl from '../../../utils/getNetworkLogoURL'
+import { getImageRequest } from '../../../utils/getRequest'
+import getSetup from '../../../utils/getSetup'
+import getToken from '../../../utils/getToken'
+
+export const onRequest: PagesFunction = async ({ params, request }) => {
+  try {
+    const origin = new URL(request.url).origin
+    const { index } = params
+    const networkName = String(index[0])
+    const tokenAddress = String(index[1])
+
+    const cacheUrl = origin + '/tokens/' + networkName + '/' + tokenAddress
+
+    const data = await getImageRequest(cacheUrl, () => getToken(networkName, tokenAddress, cacheUrl))
+
+    if (!data) {
+      return new Response('Asset not found.', { status: 404 })
+    }
+
+    data.ogImage = data.ogImage ?? origin + '/images/192x192_App_Icon.png'
+    data.name = data.name ?? 'Token'
+    data.symbol = data.symbol ?? 'UNK'
+
+    const setupPromise = getSetup()
+    const palettePromise = getColor(data.ogImage)
+
+    const [setup, palette] = await Promise.all([setupPromise, palettePromise])
+    const { fontData, watermark } = setup
+
+    const networkLogo = getNetworkLogoUrl(networkName.toUpperCase())
+
+    // Capitalize name such that each word starts with a capital letter
+    const words = data.name.split(' ')
+    let name = ''
+    for (let i = 0; i < words.length; i++) {
+      name += words[i].charAt(0).toUpperCase() + words[i].slice(1).toLowerCase() + ' '
```

Comment:
i'm all for for loops, but this is a pretty nice case of `.map` being cleaner, don't have to change it tho